File: JavaScript/AugmentingModules/student.js
var world = world || {};

world.Student = (function (w) {
    "use strict";
    function Student(name, age, email, course) {
        w.Person.call(this, name, age, email);
        this._course = course;
    }

    Student.extend(w.Person);

    Student.prototype.toString = function () {
        return w.Person.prototype.toString.call(this) + ', Course: ' + this._course;
    };

    return Student;
}(world));

File: Java/tryReflection/Student.java
package tryReflection;

public class Student {
	private String name;
	private int age;
	private int favoriteNumber;

	public Student(String name, int age) {
		this.name = name;
		this.age = age;
		this.favoriteNumber = this.age + (int) this.name.charAt(0);
	}

	@Override
	public String toString() {
		return "Student name = " + name + ", age = " + age + ", favorite number = "+favoriteNumber;
	}
}

